index.js

```diff
@@ -25,6 +26,71 @@ module.exports = {
     ...readdirSync(join(__dirname, 'rules')).map(rule => join(__dirname, 'rules', rule))
   ],
   overrides: [
+    {
+      files: ['*.js', '*.ts'],
+      rules: {
+        'node/no-unsupported-features/es-syntax': [
+          "off"
+        ]
+      }
+    },
+    {
+      files: [
+        '*.ts'
+      ],
+      parser: '@typescript-eslint/parser',
```

Comment:
@kunagpal moved the parser declaration here to prevent any unexpected issues with JS files.